### PR TITLE
Automatically crop scanned pictures

### PR DESCRIPTION
**Pull request by TheGrayStone**

### CODE REVIEW

**Comment by TheGrayStone** [comment]
One thing still missing: compiled version as .exe

File: projects/bp-gallery/src/views/gallery/shared/helpers/scanner-hook.ts [hunk 1]
```diff
@@ -14,6 +15,8 @@ const useScanner = () => {
   const [scanners, setScanners] = useState<string[]>();
   const [selectedScannerId, setSelectedScannerId] = useState<number>(0);
 
+  const [shouldCrop, setShouldCrop] = useState<boolean>(false);
```
Comment (BenediktV) [inline projects/bp-gallery/src/views/gallery/shared/helpers/scanner-hook.ts:18]: Maybe useful but I'm not sure how important: Persisting the shouldCrop state e.g. via the localStorage

**Review by BenediktV — APPROVED** [review]
🚀

File: projects/bp-gallery/src/views/gallery/shared/helpers/scanner-hook.ts [hunk 2]
```diff
@@ -26,6 +29,13 @@ const useScanner = () => {
     [socket]
   );
 
+  const toggleAutoCrop = useCallback(() => {
+    setShouldCrop(shouldCrop => {
+      socket.send(`${COMMANDS.SET_CROPPING} ${shouldCrop ? '0' : '1'}`);
+      return !shouldCrop;
+    });
```
Comment (Olliwehr) [inline projects/bp-gallery/src/views/gallery/shared/helpers/scanner-hook.ts:36]:
```suggestion
    setShouldCrop(shouldCrop => {
      const toggledShouldCrop = !shouldCrop;
      socket.send(`${COMMANDS.SET_CROPPING} ${toggledShouldCrop ? '1' : '0'}`);
      return toggledShouldCrop;
    });
```

I think this makes it clearer for me what is going on here. At first i was very confused why you sent a '0' (for false) to the Websocket when `shouldCrop` has a truthy value, but then i realized you implemented a toggle right? 😅

File: projects/bp-gallery/src/views/gallery/shared/helpers/scanner-hook.ts (hunk 1 above)
Comment (Olliwehr) [reply projects/bp-gallery/src/views/gallery/shared/helpers/scanner-hook.ts:18]: Hmm, if we want to enforce that he spends a bit more time on each picture, i think it is quite acceptable that he needs to manually configure it every time he scans 😅

**Review by Olliwehr — COMMENTED** [review]
That looks like you are a real pythonist! 💰

**Pull request merged**